code/core/src/main/java/com/adobe/marketing/mobile/internal/DataMarshaller.kt

```diff
@@ -70,9 +70,13 @@ internal object DataMarshaller {
                 NOTIFICATION_IDENTIFIER_KEY -> LOCAL_NOTIFICATION_ID_KEY
                 else -> key
             }
-            val value = extraBundle[key]
-            if (value?.toString()?.isNotEmpty() == true) {
-                marshalledData[newKey] = value
+            try {
+                val value = extraBundle[key]
+                if (value?.toString()?.isNotEmpty() == true) {
+                    marshalledData[newKey] = value
+                }
+            } catch (e: Exception) {
+                Log.error(CoreConstants.LOG_TAG, LOG_TAG, "Failed to retrieve data (key = $key) from Activity, error is: ${e.message}")
```

Comment:
I think we can to test this flow without mocks as follows : 

```
    class ObjectThrowsOnToString : Serializable {
        override fun toString(): String {
            throw IllegalStateException("This is a test exception")
        }
    }

    @Test
    fun testMarshalIntentExtras_CatchesExceptionsDuring() {
        val throwsException = ObjectWithToString()

        val intent = Intent(ApplicationProvider.getApplicationContext(), TestActivity::class.java).apply {
            // Add an extra to the bundle with a key that will cause an exception when trying to retrieve it
            putExtra("testKey", throwsException)
        }

        val activity = activityTestRule.launchActivity(intent)

        try {
            val returned = DataMarshaller.marshal(activity)
            // check returned map here
        } catch (e: Exception) {
            fail("Exception should have been caught")
        }
    }

```

Reply:
@prudrabhat @navratan-soni thank you for the suggestions. I tried the "without mocks" method, and it works great.